homework/index.js

```diff
@@ -29,19 +29,95 @@
+    data.forEach(elem => {
+      createAndAppend('option', select, { value: data.indexOf(elem), text: elem.name });
+    });
+    // Left Column
+    // const container = createAndAppend('div', root, { id: 'container' });
+    const leftdiv = createAndAppend('div', root, { id: 'leftdiv' });
+    const table = createAndAppend('table', leftdiv);
+    for (let i = 0; i < 4; i++) {
+      const tr = createAndAppend('tr', table);
+      const theads = ['Repository:', 'Description:', 'Forks:', 'Updated:'];
+      createAndAppend('th', tr, { id: `th${i}`, text: theads[i] });
+      const tds = ['name', 'description', 'forks', 'updated_at'];
+      if (tds[i] === 'name') {
+        const linked = createAndAppend('td', tr, { id: `td${i}` });
+        createAndAppend('a', linked, {
+          id: 'linked',
+          text: data[0][tds[i]],
+          href: data[0].html_url,
+          target: '_blank',
+        });
+      } else {
+        createAndAppend('td', tr, { id: `td${i}`, text: data[0][tds[i]] });
+      }
+    }
+
+    select.addEventListener('change', event => {
+      const ahref = document.getElementById('linked');
+      ahref.textContent = data[event.target.value].name;
+      ahref.href = data[event.target.value].html_url;
+      const td1 = document.getElementById('td1');
+      td1.textContent = data[event.target.value].description;
+      const td2 = document.getElementById('td2');
+      td2.textContent = data[event.target.value].forks;
+      const td3 = document.getElementById('td3');
+      td3.textContent = data[event.target.value].updated_at;
+    });
+    // right column
+    const rightdiv = createAndAppend('div', root, { id: 'rightdiv' });
+    createAndAppend('p', rightdiv, { id: 'rightp', text: 'Contributors' });
+    const ul = createAndAppend('ul', rightdiv);
+    function contributors() {
+      fetchJSON(data[select.value].contributors_url, (err, dt) => {
+        if (err) {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        } else {
+          removeChildren(ul);
+          for (let i = 0; i < dt.length; i++) {
+            const href = createAndAppend('a', ul, { href: dt[i].html_url, target: '_blank' });
+            const li = createAndAppend('li', href, { id: `li${i}` });
+            createAndAppend('img', li, { src: dt[i].avatar_url });
+            const lidiv = createAndAppend('div', li, { class: 'contributor_div' });
+            createAndAppend('p', lidiv, { text: dt[i].login });
+            createAndAppend('p', lidiv, { text: dt[i].contributions });
+          }
+        }
+      });
+    }
+    contributors();
+    select.onchange = contributors;
+  }
 
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
```

Comment:
Because you are making new functions here you are not using the `root` const you made, which is giving you an error on the linter check. You should run `npm run lint` before you make a pull request to find these little mistakes.

Reply:
i am using root inside the 'if' for appending 'div'. But anyway i deleted it in second week homework.

Reply:
it is done 👍